From a2a30da8c1db0796a8e9e72d1072c49db716b5c0 Mon Sep 17 00:00:00 2001
From: Javier Pena <jpena@redhat.com>
Date: Thu, 3 Dec 2015 16:28:55 +0100
Subject: [PATCH] Allow creating an Opt with type=str

Commit 545dd4113e62f8f4aa8b6f147e7c80ce8d460975 broke config file
generation for a number of projects. The reason for this is that
config options created by keystoneauth with type=str (see
https://github.com/openstack/keystoneauth/blob/master/keystoneauth1/loading/opts.py#L68)
did not have a format_defaults method.

Class Opt supports creating objects with type=None, which are
converted to types.String. We can extend this support to type=str.

Change-Id: I4a40a409cd900d166701e1860daea1def6ce4e64
---
 oslo_config/cfg.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/oslo_config/cfg.py b/oslo_config/cfg.py
index 6819499..75c652d 100644
--- a/oslo_config/cfg.py
+++ b/oslo_config/cfg.py
@@ -702,7 +702,7 @@ class Opt(object):
             raise ValueError('illegal name %s with prefix _' % (name,))
         self.name = name
 
-        if type is None:
+        if type in [None, str]:
             type = types.String()
 
         if not callable(type):
-- 
2.4.3

